packages/app/components/strategies/Strategies.tsx

```diff
@@ -1,18 +1,16 @@
 "use client";
 
 import { Tab } from "@headlessui/react";
-import { useNetwork } from "wagmi";
 
-import { ChainId } from "@stackly/sdk";
-import { Strategy } from "@/contexts";
+import { Strategy, useStackboxFormContext } from "@/contexts";
 import { tabButtonStyles } from "@/components";
 
 import { StrategyCard } from "./StrategyCard";
 import { STRATEGY_CATEGORIES } from "./constants";
 
 export const Strategies = () => {
-  const { chain } = useNetwork();
-  const chainId = chain?.id ?? ChainId.GNOSIS;
+  const { stackboxFormState } = useStackboxFormContext();
+  const [chainId] = stackboxFormState.chainIdState;
```

Comment:
I'm wondering if this makes sense. Why are we storing the chainId on `stackboxFormState` ?

Reply:
We need a single source of truth for the ChainID, and we don't have a ChainContext, that's why I decided to place it in the Stackbox Form context.